test/libsolidity/semanticTests/functionCall/call_options_overload.sol

```diff
@@ -0,0 +1,12 @@
+contract C {
+    function f(uint x) external payable returns (uint) { return 1; }
+    function f(uint x, uint y) external payable returns (uint) { return 2; }
+    function call() public payable returns (uint x, uint y) {
+        x = this.f{value: 10}(2);
+        y = this.f{gas: 1000}(2, 3);
+    }
```

Comment:
Can you also add a mixed case (`f{value:10, gas:100}`) ?

Reply:
oh come on - tests are always written keeping the actual implementation in mind - if you just randomly test stuff it won't help you anyway and the implementation obviously shows that you can have anything inside there even `this.f{value: f(2)}(2)`.

Reply:
Is it a sane approach to only consider the implementation and not what people may end up doing?